dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCClient.scala

```diff
@@ -0,0 +1,1056 @@
+
+        // Construct all CETs
+        val (cetWinLocalF, cetWinLocalWitness) = createCETWin(winSig)
+        val (cetLoseLocalF, cetLoseLocalWitness) = createCETLose(loseSig)
+        val cetWinRemoteF = createCETWinRemote()
+        val cetLoseRemoteF = createCETLoseRemote()
+        val refundTxF = createRefundTx(refundSig)
+
+        for {
+          cetWinLocal <- cetWinLocalF
+          cetLoseLocal <- cetLoseLocalF
+          (cetWinRemote, remoteWinSig, cetWinRemoteWitness) <- cetWinRemoteF
+          (cetLoseRemote, remoteLoseSig, cetLoseRemoteWitness) <- cetLoseRemoteF
+          (refundTx, remoteRefundSig) <- refundTxF
+          localFundingSigs <- createFundingTransactionSigs()
+          _ <- {
+            if (winIsFirst) {
+              sendSigs(remoteRefundSig,
+                       remoteWinSig,
+                       remoteLoseSig,
+                       localFundingSigs)
+            } else {
+              sendSigs(remoteRefundSig,
+                       remoteLoseSig,
+                       remoteWinSig,
+                       localFundingSigs)
+            }
+          }
+          fundingTx <- getFundingTx
+        } yield {
+          SetupDLC(
+            fundingTx,
+            cetWinLocal,
+            cetWinLocalWitness,
+            cetLoseLocal,
+            cetLoseLocalWitness,
+            cetWinRemote.txIdBE,
+            cetWinRemoteWitness,
+            cetLoseRemote.txIdBE,
+            cetLoseRemoteWitness,
+            refundTx
+          )
+        }
+    }
+  }
+
+  def constructClosingTx(
+      privKey: ECPrivateKey,
+      spendingInfo: BitcoinUTXOSpendingInfoFull,
+      isWin: Boolean,
+      spendsToLocal: Boolean): Future[Transaction] = {
+    // If spendsToLocal, use payout as value, otherwise subtract fee
+    val spendingTxF = if (spendsToLocal) {
+      val payoutValue = if (isWin) {
+        winPayout
+      } else {
+        losePayout
+      }
+
+      val txBuilder = BitcoinTxBuilder(
+        destinations = Vector(
+          TransactionOutput(payoutValue,
+                            P2WPKHWitnessSPKV0(privKey.publicKey))),
+        utxos = Vector(spendingInfo),
+        feeRate = feeRate,
+        changeSPK = emptyChangeSPK,
+        network = network
+      )
+
+      txBuilder.flatMap(_.sign)
+    } else {
+      val txBuilder = BitcoinTxBuilder(
+        destinations = Vector(
+          TransactionOutput(spendingInfo.output.value,
+                            P2WPKHWitnessSPKV0(privKey.publicKey))),
+        utxos = Vector(spendingInfo),
+        feeRate = feeRate,
+        changeSPK = emptyChangeSPK,
+        network = network
+      )
+
+      txBuilder.flatMap(subtractFeeAndSign)
+    }
+
+    spendingTxF.foreach(tx => logger.info(s"Closing Tx: ${tx.hex}"))
+
+    spendingTxF
+  }
+
+  /** Constructs and executes on the unilateral spending branch of a DLC
+    *
+    * @return Each transaction published and its spending info
+    */
+  def executeUnilateralDLC(
```

Comment:
important change: execution of unilateral DLC is not split into local and remote methods, one for each output. (note that the remote case is used in justice case as well)

Reply:
Link to the dlcspec defining what the unlilateral dlc execution is